martian/src/stage.rs

```diff
@@ -478,6 +489,18 @@ impl MartianRover {
     pub fn pipelines_version(&self) -> String {
         self.version.pipelines.clone()
     }
+
+    /// Add a message to the martian alarm system.
+    /// If this rover was not initialized with metadata, such as in test mode,
+    /// log at warning level instead.
+    pub fn alarm(&mut self, message: &str) -> Result<(), Error> {
+        if let Some(md) = &self.metadata {
+            md.borrow_mut().alarm(message)
```

Comment:
I'm a little confused about why this needs a mutable borrow.

Reply:
`alarm`, deep inside, calls `update_journal_main` which is `&mut self` because it may update an internal file cache that I think tracks whether or not certain files have been created.

Reply:
Ok, digging into this, it looks like the reason why it needs to be mutable is because writing a file updates its cache object:
https://github.com/martian-lang/martian-rust/blob/547847f7eb407a7a3f2b89f1b22a0d3be27f1d42/martian/src/metadata.rs#L206

But... I kind of feel like that cache is maybe a premature optimization. Maybe it would be better to just not have that cache, so that the method wouldn't require a mutable reference.  Either that or somehow isolate the mutability to the cache reference.

Reply:
I'm not sure it even really matters, though?  I just realized that the actual final method call `MartianRover::alarm` doesn't need to be `&mut self` because of the run-time borrow checking provided by `RefCell`.  `MartianRover` isn't `Clone`, so there's only ever two references to the `Metadata` floating around and really only a single one in scope at any time.

Reply:
Since this doesn't change the API, we can come back in follow-up work and further refactor if desired.  My main goal here is to allow using the `alarm` mechanism.

Reply:
IF it doesn't need to be `&mut self`, then neither do most of the other things on there.

Reply:
The method call `Metadata::alarm` needs to be `&mut self`.  The method call `MartianRover::alarm` only needs `&self` because `RefCell` is providing runtime borrow checking.

Reply:
Basically I'm still confused about why we need the `RefCell`.  That whole thing seems unnecessary.